fhir-persistence-schema/src/test/java/com/ibm/fhir/schema/prior/FhirResourceTableGroup455.java

```diff
@@ -0,0 +1,931 @@
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.PARAMETER_NAME_ID;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.PATIENT_CURRENT_REFS;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.PATIENT_LOGICAL_RESOURCES;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.PK;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.QUANTITY_VALUE;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.QUANTITY_VALUE_HIGH;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.QUANTITY_VALUE_LOW;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.REF_VERSION_ID;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.RESOURCE_ID;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.RESOURCE_TOKEN_REFS;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.RESOURCE_TYPES;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.RESOURCE_TYPE_ID;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.STR_VALUE;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.STR_VALUE_LCASE;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.TOKEN_VALUE;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.TOKEN_VALUES_V;
+import static com.ibm.fhir.schema.control.FhirSchemaConstants.VERSION_ID;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import com.ibm.fhir.database.utils.api.IDatabaseStatement;
+import com.ibm.fhir.database.utils.common.AddColumn;
+import com.ibm.fhir.database.utils.common.AddForeignKeyConstraint;
+import com.ibm.fhir.database.utils.common.DropColumn;
+import com.ibm.fhir.database.utils.common.DropForeignKeyConstraint;
+import com.ibm.fhir.database.utils.common.DropIndex;
+import com.ibm.fhir.database.utils.model.AlterTableIdentityCache;
+import com.ibm.fhir.database.utils.model.ColumnBase;
+import com.ibm.fhir.database.utils.model.ColumnDefBuilder;
+import com.ibm.fhir.database.utils.model.CreateIndex;
+import com.ibm.fhir.database.utils.model.ForeignKeyConstraint;
+import com.ibm.fhir.database.utils.model.Generated;
+import com.ibm.fhir.database.utils.model.GroupPrivilege;
+import com.ibm.fhir.database.utils.model.IDatabaseObject;
+import com.ibm.fhir.database.utils.model.ObjectGroup;
+import com.ibm.fhir.database.utils.model.PhysicalDataModel;
+import com.ibm.fhir.database.utils.model.SessionVariableDef;
+import com.ibm.fhir.database.utils.model.Table;
+import com.ibm.fhir.database.utils.model.Tablespace;
+import com.ibm.fhir.database.utils.model.View;
+import com.ibm.fhir.schema.control.FhirSchemaConstants;
+import com.ibm.fhir.schema.control.FhirSchemaTags;
+import com.ibm.fhir.schema.control.FhirSchemaVersion;
+
+/**
+ * Utility to create all the tables associated with a particular resource type
+ */
+public class FhirResourceTableGroup455 {
```

Comment:
It might be a bit cleaner just to download the fhir-persistence-schema cli jar (version 4.5.5) and execute it like an external program

Reply:
I had originally put in a placeholder hook in the fhir-persistence-schema's pom.xml for doing just that:
```
<plugin>
    <artifactId>exec-maven-plugin</artifactId>
    <version>1.6.0</version>
    <groupId>org.codehaus.mojo</groupId>
    <executions>
        <execution>
            <id>deployPriorVersions</id>
            <phase>test</phase>
            <goals>
                <goal>exec</goal>
            </goals>
            <configuration>
                <executable>bash</executable>
                <commandlineArgs>deployPriorVersions.sh</commandlineArgs>
            </configuration>
        </execution>
    </executions>
</plugin>
```

However, at the time the fhir-persistence-schema didn't support the instantiation of derby dbs, and so I had to revert to this pattern where I copy-pasted the old FhirResourceTableGroup into the project as a test dependency.

Reply:
Worth discussing, especially if we decide to make a breaking change to our schema model apis. But I do like the idea of the build being standalone and not breaking when some old release gets purged from a repo.